nestjs/src/courses/task-verifications/self-education.service.ts

```diff
@@ -63,19 +92,33 @@ export class SelfEducationService {
       },
     } = courseTask.task.attributes as SelfEducationAttributes;
 
-    const { id: courseTaskId, type: courseTaskType, maxScore } = courseTask;
+    if (studentAnswers.length !== numberOfQuestions) {
+      throw new BadRequestException(
+        `Number of submitted answers (${studentAnswers.length}) does not match the number of questions (${numberOfQuestions})`,
+      );
+    }
 
-    const verifications = await this.taskVerificationsRepository.find({
-      where: { studentId, courseTaskId },
-      order: { createdDate: 'DESC' },
-    });
+    // Check if all answer values are integers
+    if (studentAnswers.flatMap(a => (Array.isArray(a.value) ? a.value : [a.value])).some(a => !Number.isInteger(a))) {
+      throw new BadRequestException('Invalid answer value');
+    }
+
```

Comment:
```suggestion

    if (studentAnswers.some(a => !Number.isInteger(a.index))) {
      throw new BadRequestException('Invalid answer index');
    }

```
Because everything can be sent by direct post request.
I am not sure btw whether FE sends integers, probably it may send strings like ['1', '2', 3'].
And it will be not valid then.

Reply:
added. thanks